components/mediation-connector/org.wso2.carbon.connector.core/src/main/java/org/wso2/carbon/connector/core/RefreshAccessToken.java

```diff
@@ -0,0 +1,120 @@
+        propertyKeySet.remove("Accept-Encoding");
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("Start : Salesforce Refresh Access Token mediator. If you want to observe refresh " +
+                    "access token header and wire logs please enable Apache HTTP Client logs in Log4j.properties");
+
+            if (synLog.isTraceTraceEnabled()) {
+                synLog.traceTrace("Message : " + messageContext.getEnvelope());
+            }
+        }
+
+        String refreshUrl;
+        String customRefreshUrl = (String) messageContext.getProperty("uri.var.customRefreshUrl");
+
+        if (StringUtils.isNotEmpty(customRefreshUrl)) {
+            refreshUrl = customRefreshUrl;
+        } else {
+            StringBuilder urlStringBuilder = new StringBuilder();
+            urlStringBuilder.append(messageContext.getProperty("uri.var.hostName"));
+            urlStringBuilder.append("/services/oauth2/token?grant_type=refresh_token");
+            String clientId = (String) messageContext.getProperty("uri.var.clientId");
+            if (StringUtils.isNotEmpty(clientId)) {
+                urlStringBuilder.append("&client_id=").append(clientId);
+            }
+            String clientSecret = (String) messageContext.getProperty("uri.var.clientSecret");
+            if (StringUtils.isNotEmpty(clientSecret)) {
+                urlStringBuilder.append("&client_id=").append(clientSecret);
+            }
+            urlStringBuilder.append("&refresh_token=").append(messageContext.getProperty("uri.var.refreshToken"));
+            urlStringBuilder.append("&format=json");
+            refreshUrl = urlStringBuilder.toString();
+        }
+
+        HttpGet httpget = new HttpGet(refreshUrl);
+        CloseableHttpResponse httpResponse = null;
+
+        try {
+            httpResponse = httpClient.execute(httpget);
+            Scanner scanner = new Scanner(httpResponse.getEntity().getContent());
+            String jsonResponse = scanner.nextLine();
+            JSONObject jsonObject = new JSONObject(jsonResponse);
+
+            String accessToken = jsonObject.getString("access_token");
+            messageContext.setProperty("uri.var.accessToken", accessToken);
+
+            String instanceUrl = jsonObject.getString("instance_url");
+            messageContext.setProperty("uri.var.apiUrl", instanceUrl);
+
+            String systemTime = Long.toString(System.currentTimeMillis());
+            String newAccessRegistryPath = (String) messageContext.getProperty("uri.var.accessTokenRegistryPath");
+            String newTimeRegistryPath = (String) messageContext.getProperty("uri.var.timeRegistryPath");
+
+            if(StringUtils.isNotEmpty(accessToken)) {
+                if (!messageContext.getConfiguration().getRegistry().isResourceExists(newAccessRegistryPath)) {
+                    messageContext.getConfiguration().getRegistry().newResource(newAccessRegistryPath, false);
```

Comment:
can't we assign messageContext.getConfiguration().getRegistry() to an object and re use